lesson 5

Ja0nix opened the pull request.

Ja0nix committed:
initial

Ja0nix committed:
поиск по названиям продуктов

Поиск выводит в консоль массив из всех найденных объектов. Поиск идет по полю с названием продукта.

Ja0nix committed:
вывод результатов поиска на страницу

при поиске выводится блок с результатами перед каталогом.

Om1X commented on lesson 5/js/main.js:
```diff
@@ -0,0 +1,48 @@
+      // фильтруем продукты по названию на наличие текста из поисковой строки
+      // и выводим массив из найденных объектов
+      this.searchResult = this.products.filter((el) =>
+        el.product_name.toLowerCase().indexOf(searchText.toLowerCase()) > -1
```
поиск принято делать через регулярки и мы уже делали это. нужно было всего лишь перенести ее сюда.

Ja0nix replied:
да, не сообразила сразу, попробую с регуляркой сделать

Om1X commented on lesson 5/index.html:
```diff
@@ -0,0 +1,52 @@
+        <div class="cart">
+            <form action="#" class="search-form">
+                <input v-model="searchText" type="text" class="search-field">
+                <button class="btn-search" type="submit" @click="filterGoods(searchText), show=!show">
```
не совсем понял зачем тут переключатель show. ведь он будет меняться при пажом срабатывании события.

Om1X replied:
нет смысла передавать на вход то, к чему итак можно получить доступ внутри метода.

Ja0nix replied:
Да, точно, поняла, исправлю!

Om1X commented on lesson 5/index.html:
```diff
@@ -0,0 +1,52 @@
+                    <i class="fas fa-search"></i>
+                </button>
+            </form>
+            <button class="btn-cart" type="button">Корзина</button>
```
нет корзины.

Ja0nix replied:
сегодня доделала корзину

Ja0nix committed:
вывод корзины

Готово:
- вывод корзины по клику
- добавление товара в массив корзины
- вывод общего количества товаров в корзине на плашке в меню

Ja0nix committed:
небольшие правки стиля

Ja0nix committed:
Изменение searchText-> searchLine

по заданию другое название у поля, не внимательно сразу прочла

Ja0nix committed:
исправлена часть замечний

нет смысла передавать на вход то, к чему итак можно получить доступ внутри метода. - исправила

поиск принято делать через регулярки и мы уже делали это. нужно было всего лишь перенести ее сюда. - исправила

Ja0nix committed:
исправлен вывод поиска

Убрала переключатель show и через стили реализовала показ блока с результатами поиска
